src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -83,6 +83,27 @@ public static void AddProject(this SlnFile slnFile, string fullProjectPath, ILis
 
                 if (solutionFolders != null)
                 {
+                    if (solutionFolders.Any())
+                    {
+                        // Check for any existing projects if there is a new matching solution folder and add them to their own solution folder
+                        var duplicateProjects = slnFile.Projects.Where(p => solutionFolders.Contains(p.Name)
+                                                                        && p.TypeGuid != ProjectTypeGuids.SolutionFolderGuid).ToList();
+                        foreach (SlnProject duplicateProject in duplicateProjects)
+                        {
+                            slnFile.AddSolutionFolders(duplicateProject, new List<string>() { Path.GetDirectoryName(duplicateProject.FilePath) });
+                        }
+                    }
+                    else
+                    {
+                        // If the project conflicts with a solution folder, add it's own folder as a solution folder
+                        var duplicateProject = slnFile.Projects.Where(p => string.Equals(p.Name, slnProject.Name, StringComparison.OrdinalIgnoreCase)
+                                                                       && p.TypeGuid == ProjectTypeGuids.SolutionFolderGuid).FirstOrDefault();
+                        if (duplicateProject != null)
+                        {
+                            slnFile.AddSolutionFolders(slnProject, new List<string>() { Path.GetDirectoryName(relativeProjectPath) });
+                        }
+                    }
+
                     slnFile.AddSolutionFolders(slnProject, solutionFolders);
```

Comment:
I was confused at first as to why we did this line even if we added to the folder above, so I think this is helpful
```suggestion
                   // Even if we added a solution folder above, we still need to add the folder specified by the command option.
                    slnFile.AddSolutionFolders(slnProject, solutionFolders);
```